Build failures on Fedora Rawhide

I just ran a build of Cherrytree v1.1.4 in Fedora Copr. It built fine on F39 and F40, but Rawhide failed. The issue seems to be due to a change in the fmt-devel package from 10.2.1 in F40 to 11.0.2 in Rawhide. 

Relevant build log snippet:

~~~~
[ 77%] Building CXX object src/ct/CMakeFiles/cherrytree_shared.dir/ct_actions_export.cc.o
In file included from /usr/include/fmt/format.h:41,
                 from /usr/include/fmt/core.h:5,
                 from /usr/include/spdlog/fmt/fmt.h:28,
                 from /usr/include/spdlog/common.h:50,
                 from /usr/include/spdlog/spdlog.h:12,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_logging.h:27,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_misc_utils.h:27,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_main_win.h:33,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_actions.h:26,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_actions_export.cc:24:
/usr/include/fmt/base.h: In instantiation of â€˜static void fmt::v11::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = Glib::ustring; Formatter = fmt::v11::formatter<Glib::ustring>; Context = fmt::v11::context; typename Context::parse_context_type = fmt::v11::basic_format_parse_context<char>]â€™:
/usr/include/fmt/base.h:1383:19:   required from â€˜fmt::v11::detail::value<Context>::value(T&) [with T = const Glib::ustring; Context = fmt::v11::context]â€™
 1383 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1384 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1647:41:   required from â€˜constexpr fmt::v11::detail::value<Context> fmt::v11::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v11::context; T = const Glib::ustring; typename std::enable_if<PACKED, int>::type <anonymous> = 0]â€™
 1647 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/usr/include/fmt/base.h:2018:74:   required from â€˜constexpr fmt::v11::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v11::make_format_args(T& ...) [with Context = context; T = {const Glib::ustring, const Glib::ustring}; long unsigned int NUM_ARGS = 2; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 255; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]â€™
 2018 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2019 |       args)...}};
      |       ~~~~~                                                               
/usr/include/fmt/format.h:4365:44:   required from â€˜std::string fmt::v11::format(format_string<T ...>, T&& ...) [with T = {const Glib::ustring&, const Glib::ustring&}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, const Glib::ustring&, const Glib::ustring&>]â€™
 4365 |   return vformat(fmt, fmt::make_format_args(args...));
      |                       ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_misc_utils.h:367:23:   required from â€˜std::string str::format(const std::string&, const Args& ...) [with Args = {Glib::ustring, Glib::ustring}; std::string = std::__cxx11::basic_string<char>]â€™
  367 |     return fmt::format(str::replace(in_str, "%s", "{}").c_str(), args...);
      |            ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_actions_export.cc:137:38:   required from here
  137 |                 message = str::format(_("A folder '%s' already exists in '%s'.\n<b>Do you want to remove it?</b>"),
      |                           ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  138 |                     str::xml_escape(Glib::path_get_basename(new_filepath)), str::xml_escape(Glib::path_get_dirname(new_filepath)));
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1402:29: error: passing â€˜const fmt::v11::formatter<Glib::ustring>â€™ as â€˜thisâ€™ argument discards qualifiers [-fpermissive]
 1402 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_logging.h:42:8: note:   in call to â€˜auto fmt::v11::formatter<Glib::ustring>::format(Glib::ustring, FormatContext&) [with FormatContext = fmt::v11::context]â€™
   42 |   auto format(Glib::ustring c, FormatContext& ctx) {
      |        ^~~~~~
In file included from /usr/include/fmt/format.h:41,
                 from /usr/include/fmt/core.h:5,
                 from /usr/include/spdlog/fmt/fmt.h:28,
                 from /usr/include/spdlog/common.h:50,
                 from /usr/include/spdlog/spdlog.h:12,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_logging.h:27,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_misc_utils.h:27,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_main_win.h:33,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_actions.h:26,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_actions_edit.cc:24:
/usr/include/fmt/base.h: In instantiation of â€˜static void fmt::v11::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = Glib::ustring; Formatter = fmt::v11::formatter<Glib::ustring>; Context = fmt::v11::context; typename Context::parse_context_type = fmt::v11::basic_format_parse_context<char>]â€™:
/usr/include/fmt/base.h:1383:19:   required from â€˜fmt::v11::detail::value<Context>::value(T&) [with T = const Glib::ustring; Context = fmt::v11::context]â€™
 1383 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1384 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1647:41:   required from â€˜constexpr fmt::v11::detail::value<Context> fmt::v11::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v11::context; T = const Glib::ustring; typename std::enable_if<PACKED, int>::type <anonymous> = 0]â€™
 1647 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/usr/include/fmt/base.h:2018:74:   required from â€˜constexpr fmt::v11::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v11::make_format_args(T& ...) [with Context = context; T = {const Glib::ustring}; long unsigned int NUM_ARGS = 1; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 15; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]â€™
 2018 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2019 |       args)...}};
      |       ~~~~~                                                               
/usr/include/fmt/format.h:4365:44:   required from â€˜std::string fmt::v11::format(format_string<T ...>, T&& ...) [with T = {const Glib::ustring&}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, const Glib::ustring&>]â€™
 4365 |   return vformat(fmt, fmt::make_format_args(args...));
      |                       ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_misc_utils.h:367:23:   required from â€˜std::string str::format(const std::string&, const Args& ...) [with Args = {Glib::ustring}; std::string = std::__cxx11::basic_string<char>]â€™
  367 |     return fmt::format(str::replace(in_str, "%s", "{}").c_str(), args...);
      |            ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_actions_edit.cc:319:45:   required from here
  319 |         throw std::runtime_error(str::format(_("Failed to retrieve the content of the node '%s'"), node.get_node_name()));
      |                                  ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1402:29: error: passing â€˜const fmt::v11::formatter<Glib::ustring>â€™ as â€˜thisâ€™ argument discards qualifiers [-fpermissive]
 1402 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4-build/cherrytree-1.1.4/src/ct/ct_logging.h:42:8: note:   in call to â€˜auto fmt::v11::formatter<Glib::ustring>::format(Glib::ustring, FormatContext&) [with FormatContext = fmt::v11::context]â€™
   42 |   auto format(Glib::ustring c, FormatContext& ctx) {
      |        ^~~~~~
make[2]: *** [src/ct/CMakeFiles/cherrytree_shared.dir/build.make:90: src/ct/CMakeFiles/cherrytree_shared.dir/ct_actions_export.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [src/ct/CMakeFiles/cherrytree_shared.dir/build.make:76: src/ct/CMakeFiles/cherrytree_shared.dir/ct_actions_edit.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:199: src/ct/CMakeFiles/cherrytree_shared.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
~~~~

@giuspen That fixed the error in my initial report, but I'm seeing a similar error on the next step

~~~~
[ 78%] Building CXX object src/ct/CMakeFiles/cherrytree_shared.dir/ct_actions_export.cc.o
[ 79%] Building CXX object src/ct/CMakeFiles/cherrytree_shared.dir/ct_actions_format.cc.o
In file included from /usr/include/fmt/format.h:41,
                 from /usr/include/fmt/core.h:5,
                 from /usr/include/spdlog/fmt/fmt.h:28,
                 from /usr/include/spdlog/common.h:50,
                 from /usr/include/spdlog/spdlog.h:12,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821gitccc2d10-build/cherrytree-ccc2d101f24a409efddb2f29e8c14002c9836a85/src/ct/ct_logging.h:27,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821gitccc2d10-build/cherrytree-ccc2d101f24a409efddb2f29e8c14002c9836a85/src/ct/ct_misc_utils.h:27,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821gitccc2d10-build/cherrytree-ccc2d101f24a409efddb2f29e8c14002c9836a85/src/ct/ct_main_win.h:33,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821gitccc2d10-build/cherrytree-ccc2d101f24a409efddb2f29e8c14002c9836a85/src/ct/ct_actions.h:26,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821gitccc2d10-build/cherrytree-ccc2d101f24a409efddb2f29e8c14002c9836a85/src/ct/ct_actions_others.cc:24:
/usr/include/fmt/base.h: In instantiation of ‘static void fmt::v11::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = fs::path; Formatter = fmt::v11::formatter<fs::path>; Context = fmt::v11::context; typename Context::parse_context_type = fmt::v11::basic_format_parse_context<char>]’:
/usr/include/fmt/base.h:1383:19:   required from ‘fmt::v11::detail::value<Context>::value(T&) [with T = const fs::path; Context = fmt::v11::context]’
 1383 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1384 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1647:41:   required from ‘constexpr fmt::v11::detail::value<Context> fmt::v11::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v11::context; T = const fs::path; typename std::enable_if<PACKED, int>::type <anonymous> = 0]’
 1647 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/usr/include/fmt/base.h:2018:74:   required from ‘constexpr fmt::v11::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v11::make_format_args(T& ...) [with Context = context; T = {const fs::path}; long unsigned int NUM_ARGS = 1; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 15; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]’
 2018 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2019 |       args)...}};
      |       ~~~~~                                                               
/usr/include/spdlog/logger.h:328:75:   required from ‘void spdlog::logger::log_(spdlog::source_loc, spdlog::level::level_enum, spdlog::string_view_t, Args&& ...) [with Args = {const fs::path&}; spdlog::string_view_t = fmt::v11::basic_string_view<char>]’
  328 |             fmt::vformat_to(fmt::appender(buf), fmt, fmt::make_format_args(args...));
      |                                                      ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
/usr/include/spdlog/logger.h:80:13:   required from ‘void spdlog::logger::log(spdlog::source_loc, spdlog::level::level_enum, fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const fs::path&>]’
   80 |         log_(loc, lvl, details::to_string_view(fmt), std::forward<Args>(args)...);
      |         ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:85:12:   required from ‘void spdlog::logger::log(spdlog::level::level_enum, fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const fs::path&>]’
   85 |         log(source_loc{}, lvl, fmt, std::forward<Args>(args)...);
      |         ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:135:12:   required from ‘void spdlog::logger::debug(fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const fs::path&>]’
  135 |         log(level::debug, fmt, std::forward<Args>(args)...);
      |         ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/spdlog.h:163:32:   required from ‘void spdlog::debug(fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const fs::path&>]’
  163 |     default_logger_raw()->debug(fmt, std::forward<Args>(args)...);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4.20240821gitccc2d10-build/cherrytree-ccc2d101f24a409efddb2f29e8c14002c9836a85/src/ct/ct_actions_others.cc:901:26:   required from here
  901 |             spdlog::debug("embdrop {}", tmp_filepath);
      |             ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1402:29: error: passing ‘const fmt::v11::formatter<fs::path>’ as ‘this’ argument discards qualifiers [-fpermissive]
 1402 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821gitccc2d10-build/cherrytree-ccc2d101f24a409efddb2f29e8c14002c9836a85/src/ct/ct_widgets.h:26,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821gitccc2d10-build/cherrytree-ccc2d101f24a409efddb2f29e8c14002c9836a85/src/ct/ct_main_win.h:35:
/builddir/build/BUILD/cherrytree-future-1.1.4.20240821gitccc2d10-build/cherrytree-ccc2d101f24a409efddb2f29e8c14002c9836a85/src/ct/ct_filesystem.h:213:10: note:   in call to ‘auto fmt::v11::formatter<fs::path>::format(const fs::path&, FormatContext&) [with FormatContext = fmt::v11::context]’
  213 |     auto format(const fs::path& p, FormatContext& ctx) {
      |          ^~~~~~
make[2]: *** [src/ct/CMakeFiles/cherrytree_shared.dir/build.make:146: src/ct/CMakeFiles/cherrytree_shared.dir/ct_actions_others.cc.o] Error 1
~~~~

@giuspen with commit 76f0030, it got a little further:

~~~~
[ 82%] Building CXX object src/ct/CMakeFiles/cherrytree_shared.dir/ct_dialogs_anch_widg.cc.o
In file included from /usr/include/fmt/format.h:41,
                 from /usr/include/fmt/core.h:5,
                 from /usr/include/spdlog/fmt/fmt.h:28,
                 from /usr/include/spdlog/common.h:50,
                 from /usr/include/spdlog/spdlog.h:12,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_logging.h:27,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_filesystem.h:30,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_config.h:31,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_config.cc:26:
/usr/include/fmt/base.h: In instantiation of ‘static void fmt::v11::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = fs::path; Formatter = fmt::v11::formatter<fs::path>; Context = fmt::v11::context; typename Context::parse_context_type = fmt::v11::basic_format_parse_context<char>]’:
/usr/include/fmt/base.h:1383:19:   required from ‘fmt::v11::detail::value<Context>::value(T&) [with T = const fs::path; Context = fmt::v11::context]’
 1383 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1384 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1647:41:   required from ‘constexpr fmt::v11::detail::value<Context> fmt::v11::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v11::context; T = const fs::path; typename std::enable_if<PACKED, int>::type <anonymous> = 0]’
 1647 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/usr/include/fmt/base.h:2018:74:   required from ‘constexpr fmt::v11::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v11::make_format_args(T& ...) [with Context = context; T = {const fs::path, const fs::path}; long unsigned int NUM_ARGS = 2; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 255; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]’
 2018 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2019 |       args)...}};
      |       ~~~~~                                                               
/usr/include/spdlog/logger.h:328:75:   required from ‘void spdlog::logger::log_(spdlog::source_loc, spdlog::level::level_enum, spdlog::string_view_t, Args&& ...) [with Args = {const fs::path&, const fs::path&}; spdlog::string_view_t = fmt::v11::basic_string_view<char>]’
  328 |             fmt::vformat_to(fmt::appender(buf), fmt, fmt::make_format_args(args...));
      |                                                      ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
/usr/include/spdlog/logger.h:80:13:   required from ‘void spdlog::logger::log(spdlog::source_loc, spdlog::level::level_enum, fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const fs::path&, const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const fs::path&, const fs::path&>]’
   80 |         log_(loc, lvl, details::to_string_view(fmt), std::forward<Args>(args)...);
      |         ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:85:12:   required from ‘void spdlog::logger::log(spdlog::level::level_enum, fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const fs::path&, const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const fs::path&, const fs::path&>]’
   85 |         log(source_loc{}, lvl, fmt, std::forward<Args>(args)...);
      |         ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:150:12:   required from ‘void spdlog::logger::error(fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const fs::path&, const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const fs::path&, const fs::path&>]’
  150 |         log(level::err, fmt, std::forward<Args>(args)...);
      |         ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/spdlog.h:178:32:   required from ‘void spdlog::error(fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const fs::path&, const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const fs::path&, const fs::path&>]’
  178 |     default_logger_raw()->error(fmt, std::forward<Args>(args)...);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_config.cc:72:26:   required from here
   72 |             spdlog::error("{} -> {}", _configFilepathTmp, _configFilepath);
      |             ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1402:29: error: passing ‘const fmt::v11::formatter<fs::path>’ as ‘this’ argument discards qualifiers [-fpermissive]
 1402 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_filesystem.h:213:10: note:   in call to ‘auto fmt::v11::formatter<fs::path>::format(const fs::path&, FormatContext&) [with FormatContext = fmt::v11::context]’
  213 |     auto format(const fs::path& p, FormatContext& ctx) {
      |          ^~~~~~
In file included from /usr/include/fmt/format.h:41,
                 from /usr/include/fmt/core.h:5,
                 from /usr/include/spdlog/fmt/fmt.h:28,
                 from /usr/include/spdlog/common.h:50,
                 from /usr/include/spdlog/spdlog.h:12,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_logging.h:27,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_filesystem.h:30,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_config.h:31,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_app.h:29,
                 from /builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_app.cc:25:
/usr/include/fmt/base.h: In instantiation of ‘static void fmt::v11::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = fs::path; Formatter = fmt::v11::formatter<fs::path>; Context = fmt::v11::context; typename Context::parse_context_type = fmt::v11::basic_format_parse_context<char>]’:
/usr/include/fmt/base.h:1383:19:   required from ‘fmt::v11::detail::value<Context>::value(T&) [with T = const fs::path; Context = fmt::v11::context]’
 1383 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1384 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1647:41:   required from ‘constexpr fmt::v11::detail::value<Context> fmt::v11::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v11::context; T = const fs::path; typename std::enable_if<PACKED, int>::type <anonymous> = 0]’
 1647 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/usr/include/fmt/base.h:2018:74:   required from ‘constexpr fmt::v11::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v11::make_format_args(T& ...) [with Context = context; T = {const char [12], const fs::path}; long unsigned int NUM_ARGS = 2; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 252; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]’
 2018 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2019 |       args)...}};
      |       ~~~~~                                                               
/usr/include/spdlog/logger.h:328:75:   required from ‘void spdlog::logger::log_(spdlog::source_loc, spdlog::level::level_enum, spdlog::string_view_t, Args&& ...) [with Args = {const char (&)[12], const fs::path&}; spdlog::string_view_t = fmt::v11::basic_string_view<char>]’
  328 |             fmt::vformat_to(fmt::appender(buf), fmt, fmt::make_format_args(args...));
      |                                                      ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
/usr/include/spdlog/logger.h:80:13:   required from ‘void spdlog::logger::log(spdlog::source_loc, spdlog::level::level_enum, fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const char (&)[12], const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const char (&)[12], const fs::path&>]’
   80 |         log_(loc, lvl, details::to_string_view(fmt), std::forward<Args>(args)...);
      |         ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:85:12:   required from ‘void spdlog::logger::log(spdlog::level::level_enum, fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const char (&)[12], const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const char (&)[12], const fs::path&>]’
   85 |         log(source_loc{}, lvl, fmt, std::forward<Args>(args)...);
      |         ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/logger.h:140:12:   required from ‘void spdlog::logger::info(fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const char (&)[12], const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const char (&)[12], const fs::path&>]’
  140 |         log(level::info, fmt, std::forward<Args>(args)...);
      |         ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/spdlog/spdlog.h:168:31:   required from ‘void spdlog::info(fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const char (&)[12], const fs::path&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const char (&)[12], const fs::path&>]’
  168 |     default_logger_raw()->info(fmt, std::forward<Args>(args)...);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_app.cc:155:29:   required from here
  155 |                 spdlog::info("{} Last doc not found: {}", __FUNCTION__, last_doc_path);
      |                 ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fmt/base.h:1402:29: error: passing ‘const fmt::v11::formatter<fs::path>’ as ‘this’ argument discards qualifiers [-fpermissive]
 1402 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/cherrytree-future-1.1.4.20240821git76f0030-build/cherrytree-76f0030e2e2b6e1488148d3828baeb8f5911eb8d/src/ct/ct_filesystem.h:213:10: note:   in call to ‘auto fmt::v11::formatter<fs::path>::format(const fs::path&, FormatContext&) [with FormatContext = fmt::v11::context]’
  213 |     auto format(const fs::path& p, FormatContext& ctx) {
      |          ^~~~~~
make[2]: *** [src/ct/CMakeFiles/cherrytree_shared.dir/build.make:258: src/ct/CMakeFiles/cherrytree_shared.dir/ct_config.cc.o] Error 1
~~~~

Here's how I setup my build, in case that's helpful:

1. `wget https://github.com/giuspen/cherrytree/archive/76f0030e2e2b6e1488148d3828baeb8f5911eb8d.zip -O ~/rpmbuild/SOURCES/76f0030e2e2b6e1488148d3828baeb8f5911eb8d.zip`
2. Edit the [spec file](https://github.com/funnelfiasco/copr-cherrytree/blob/rawhide_build_failures/cherrytree-future.spec) to use that commit hash
3. `rpmbuild -bs cherrytree-future.spec`
4. `mock -r fedora-rawhide-x86_64 ~/rpmbuild/SRPMS/cherrytree-future-1.1.4.20240821git76f0030-1.fc39.src.rpm`

--- a/src/ct/ct_export2html.cc
+++ b/src/ct/ct_export2html.cc
@@ -180,7 +180,7 @@ void CtExport2Html::nodes_all_export_to_multiple_html(bool all_tree, const CtExp
     tree_links_text += "</div>\n";
 
     // create index html page
-    Glib::ustring html_text = str::format(HTML_HEADER, _pCtMainWin->get_ct_storage()->get_file_name());
+    Glib::ustring html_text = str::format(HTML_HEADER, _pCtMainWin->get_ct_storage()->get_file_name().string());
     if (options.index_in_page) {
         html_text += "<div class='two-panels'>\n<div class='tree-panel'>\n";
         html_text += tree_links_text;
@@ -283,7 +283,7 @@ void CtExport2Html::nodes_all_export_to_single_html(bool all_tree, const CtExpor
         }
     };
 
-    Glib::ustring html_header = str::format(HTML_HEADER, _pCtMainWin->get_ct_storage()->get_file_name());
+    Glib::ustring html_header = str::format(HTML_HEADER, _pCtMainWin->get_ct_storage()->get_file_name().string());
     rFileStream->write(html_header.c_str(), html_header.bytes());
 
     // start to iterarte nodes

--- a/src/ct/ct_storage_control.cc
+++ b/src/ct/ct_storage_control.cc
@@ -78,7 +78,7 @@ static const std::string BAD_ARCHIVE{"_BAD_ARC_"};
                     return nullptr;
                 }
                 if (extracted_file_path.string() == BAD_ARCHIVE) {
-                    throw std::runtime_error(str::format(_("'%s' is Not a Valid Archive"), file_path));
+                    throw std::runtime_error(str::format(_("'%s' is Not a Valid Archive"), file_path.string()));
                 }
             }
         }
@@ -533,14 +533,14 @@ void CtStorageControl::_backupEncryptThread()
                 // move back the latest file version
                 (void)fs::move_file(pBackupEncryptData->main_backup, pBackupEncryptData->file_path);
 #if defined(DEBUG_BACKUP_ENCRYPT)
-                spdlog::debug("{} -> {}", pBackupEncryptData->main_backup, pBackupEncryptData->file_path);
+                spdlog::debug("{} -> {}", pBackupEncryptData->main_backup, pBackupEncryptData->file_path.string());
 #endif // DEBUG_BACKUP_ENCRYPT
                 _pCtMainWin->errorsDEQueue.push_back(_("Failed to encrypt the file"));
                 _pCtMainWin->dispatcherErrorMsg.emit();
                 continue;
             }
 #if defined(DEBUG_BACKUP_ENCRYPT)
-            spdlog::debug("{} => {}", pBackupEncryptData->extracted_copy, pBackupEncryptData->file_path);
+            spdlog::debug("{} => {}", pBackupEncryptData->extracted_copy, pBackupEncryptData->file_path.string());
 #endif // DEBUG_BACKUP_ENCRYPT
         }
 
@@ -729,12 +729,12 @@ void CtStorageControl::add_nodes_from_storage(const fs::path& file_path,
 {
     if (is_folder) {
         if (not fs::is_directory(file_path)) {
-            throw std::runtime_error(fmt::format("{} not a dir", file_path));
+            throw std::runtime_error(fmt::format("{} not a dir", file_path.string()));
         }
     }
     else {
         if (not fs::is_regular_file(file_path)) {
-            throw std::runtime_error(fmt::format("{} not a file", file_path));
+            throw std::runtime_error(fmt::format("{} not a file", file_path.string()));
         }
     }
 
@@ -747,7 +747,7 @@ void CtStorageControl::add_nodes_from_storage(const fs::path& file_path,
             return;
         }
         if (extracted_file_path.string() == BAD_ARCHIVE) {
-            throw std::runtime_error(str::format(_("'%s' is Not a Valid Archive"), file_path));
+            throw std::runtime_error(str::format(_("'%s' is Not a Valid Archive"), file_path.string()));
         }
     }
 

--- a/src/ct/ct_storage_multifile.cc
+++ b/src/ct/ct_storage_multifile.cc
@@ -510,7 +510,7 @@ bool CtStorageMultiFile::populate_treestore(const fs::path& dir_path, Glib::ustr
                 parsingOk = true;
             }
             catch (std::exception& ex) {
-                spdlog::error("parse {} : {} - trying first backup...", node_xml_path, ex.what());
+                spdlog::error("parse {} : {} - trying first backup...", node_xml_path.string(), ex.what());
             }
             if (not parsingOk) {
                 std::string first_backup_dir;
@@ -527,13 +527,13 @@ bool CtStorageMultiFile::populate_treestore(const fs::path& dir_path, Glib::ustr
                             parsingOk = true;
                         }
                         catch (std::exception& ex) {
-                            spdlog::error("parse {} : {} - trying backup {}...", node_xml_path, ex.what(), b+2);
+                            spdlog::error("parse {} : {} - trying backup {}...", node_xml_path.string(), ex.what(), b+2);
                         }
                         if (parsingOk) {
                             if (fs::exists(node_xml_path)) {
                                 fs::move_file(node_xml_path, node_xml_path.parent_path() / (node_xml_path.stem() + std::string{"_BAD.xml"}));
                             }
-                            spdlog::debug("parse backed up data ok, copying {} -> {}", backup_node_xml_path, node_xml_path);
+                            spdlog::debug("parse backed up data ok, copying {} -> {}", backup_node_xml_path.string(), node_xml_path.string());
                             fs::copy_file(backup_node_xml_path, node_xml_path);
                             if (error.empty()) error += _("A Restore From Backup Was Necessary For:");
                             error += "\n\n" + node_xml_path.string();

--- a/src/ct/ct_storage_sqlite.cc
+++ b/src/ct/ct_storage_sqlite.cc
@@ -207,10 +207,10 @@ void CtStorageSqlite::test_connection()
     }
     catch(std::exception& e) {
         spdlog::debug("{} {}", __FUNCTION__, e.what());
-        throw std::runtime_error(str::format(_("%s write failed - file is missing. Reattach USB drive or shared resource!"), _file_path));
+        throw std::runtime_error(str::format(_("%s write failed - file is missing. Reattach USB drive or shared resource!"), _file_path.string()));
     }
     if (not test_readwrite())
-        throw std::runtime_error(str::format(_("%s write failed - is file blocked by a sync program?"), _file_path));
+        throw std::runtime_error(str::format(_("%s write failed - is file blocked by a sync program?"), _file_path.string()));
     (void)_check_database_integrity();
 }
 
@@ -223,7 +223,7 @@ void CtStorageSqlite::try_reopen()
     }
     catch(std::exception& e) {
         spdlog::debug("{} {}", __FUNCTION__, e.what());
-        throw std::runtime_error(str::format(_("%s reopen failed - file is missing. Reattach USB drive or shared resource!"), _file_path));
+        throw std::runtime_error(str::format(_("%s reopen failed - file is missing. Reattach USB drive or shared resource!"), _file_path.string()));
     }
     (void)_check_database_integrity();
 }

